lib/active_model/serializer.rb

```diff
@@ -171,8 +186,31 @@ def attributes
       end
     end
 
+    def useless_attributes_keys
+      self.class
+        ._attributes_conditions
+        .dup
+        .each_with_object([]) do |(switch_hash, keys), useless_keys|
+        case switch_hash.keys
+        when [:if]
+          useless_keys << keys unless judge_attribute(switch_hash[:if])
+        when [:unless]
+          useless_keys << keys if judge_attribute(switch_hash[:unless])
+        else
+          useless_keys
+        end
+      end.flatten
+    end
+
     protected
 
     attr_accessor :instance_options
+
+    private
+
+    def judge_attribute(condition)
+      (condition.is_a?(Symbol) && send(condition)) ||
```

Comment:
You are right about the `case`, but I believe there should be an `else` clause with a `warn "Invalid condition #{condition} for attribute #{blah}."` in order to avoid weird error messages.